tests/conftest.py

```diff
@@ -169,3 +169,11 @@ def hypergraph1():
     H.add_nodes_from(["a", "b", "c"])
     H.add_edges_from({"e1": ["a", "b"], "e2": ["a", "b", "c"], "e3": ["c"]})
     return H
+
+
+@pytest.fixture
+def hypergraph2():
+    H = xgi.Hypergraph()
+    H.add_nodes_from([0, "b", "c"])
```

Comment:
I think the right test would be to add `["b", "c", 0]` i.e. the `0` last, so the (old) code would have overwritten `H._node[0] = H._node["b"]` and lost the neighbors of `0`. In this test, `0` is added first so it just gets relabeled to itself and there is no bug.

Reply:
The new code looks better tho and I can't spot any bugs if any.